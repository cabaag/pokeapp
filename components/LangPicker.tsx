import { Icon, Picker } from 'native-base';
import React, { useCallback, useState } from 'react';

const LangPicker: React.FC = () => {
  const [language, setLanguage] = useState('en');

  const handleChangeLang = useCallback((lang) => {
    console.log(lang);
  }, [])

  return (
    <Picker
      iosHeader="Select your language"
      iosIcon={(
        <Icon
          name="arrow-dropdown-circle"
          style={{ color: "#007aff", fontSize: 25 }}
        />
      )}
      mode="dropdown"
      onValueChange={handleChangeLang}
      selectedValue={language}
      style={{ width: undefined }}
    >
      <Picker.Item label="Wallet" value="key0" />
      <Picker.Item label="ATM Card" value="key1" />
      <Picker.Item label="Debit Card" value="key2" />
      <Picker.Item label="Credit Card" value="key3" />
      <Picker.Item label="Net Banking" value="key4" />
    </Picker>
  );
}

export default LangPicker;