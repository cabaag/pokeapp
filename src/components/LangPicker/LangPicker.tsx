import { Icon, Picker } from 'native-base';
import React, { useCallback, useState, memo } from 'react';
import { useTranslation } from 'react-i18next';

const LangPicker: React.FC = () => {
  const { t, i18n } = useTranslation();
  const [language, setLanguage] = useState(i18n.language);

  const handleChangeLang = useCallback((lang) => {
    i18n.changeLanguage(lang);
    setLanguage(lang);
  }, [])

  return (
    <Picker
      iosHeader={t("select_lang")}
      iosIcon={(
        <Icon
          name="arrow-dropdown-circle"
        />
      )}
      mode="dropdown"
      onValueChange={handleChangeLang}
      placeholder={t("select_lang")}
      selectedValue={language}
      

    >
      <Picker.Item label={t("english")} value="en" />
      <Picker.Item label={t("spanish")} value="es" />
    </Picker>
  );
}

export default memo(LangPicker);