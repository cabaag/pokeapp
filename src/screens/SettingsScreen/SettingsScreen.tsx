import * as SecureStore from 'expo-secure-store';
import { Body, Content, Form, Icon, Item, Picker, Row } from 'native-base';
import React, { memo, useEffect, useState } from 'react';
import LangPicker from '../../components/LangPicker/LangPicker';

const SettingsScreen: React.FC = () => {
  const [pokemonsPerPage, setPokemonsPerPage] = useState(50);

  useEffect(() => {
    SecureStore.getItemAsync('pokemonsPerPage').then(ppp => setPokemonsPerPage(+(ppp ?? 50)))
  }, [])

  const handlePokemonsPerPage = (value: number) => {
    setPokemonsPerPage(value);
    SecureStore.setItemAsync('pokemonsPerPage', value.toString());
  }

  return (
    <Content padder>
      <Body>
        <Row> 
          <Form>
            <Item picker>
              <LangPicker />
            </Item>
            <Item picker>
              <Picker
                iosIcon={<Icon name="arrow-down" />}
                mode="dropdown"
                onValueChange={handlePokemonsPerPage}
                placeholder="Pokemons per page"
                placeholderIconColor="#007aff"
                placeholderStyle={{ color: "#bfc6ea" }}
                selectedValue={pokemonsPerPage}
                style={{ width: undefined, color: '#000' }}
              >
                <Picker.Item label="30" value="30" />
                <Picker.Item label="50" value="50" />
                <Picker.Item label="70" value="70" />
                <Picker.Item label="100" value="100" />
              </Picker>
            </Item>
          </Form>
        </Row>
      </Body>
    </Content>
  )
}

export default memo(SettingsScreen);