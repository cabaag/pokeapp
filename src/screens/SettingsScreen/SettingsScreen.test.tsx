import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import LangPicker from '../../components/LangPicker/LangPicker';
import SettingsScreen from './SettingsScreen';


function setup(): ShallowWrapper {
  return shallow(<SettingsScreen />)
}

describe.only('<SettingsScreen />', () => {
  let wrapper: ShallowWrapper;


  beforeEach(() => {
    wrapper = setup();
  })

  it('matchs snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('contains LangPicker', () => {
    expect(wrapper.find(LangPicker)).toHaveLength(1);
  })


});