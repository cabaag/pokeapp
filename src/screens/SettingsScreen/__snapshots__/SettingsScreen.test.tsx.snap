// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsScreen /> matchs snapshot 1`] = `
<Styled(Container)>
  <Styled(Content)
    padder={true}
  >
    <Styled(Form)>
      <Styled(Item)
        picker={true}
      >
        <Styled(Label)>
          language
        </Styled(Label)>
        <Memo(LangPicker) />
      </Styled(Item)>
      <Styled(Item)
        picker={true}
      >
        <Styled(Label)>
          ppp
        </Styled(Label)>
        <Styled(PickerNB)
          iosIcon={
            <Styled(Icon)
              name="arrow-down"
            />
          }
          mode="dropdown"
          onValueChange={[Function]}
          placeholder="ppp"
          placeholderIconColor="#007aff"
          placeholderStyle={
            Object {
              "color": "#bfc6ea",
            }
          }
          selectedValue={50}
          style={
            Object {
              "color": "#000",
              "width": undefined,
            }
          }
        >
          <Item
            label="30"
            value="30"
          />
          <Item
            label="50"
            value="50"
          />
          <Item
            label="70"
            value="70"
          />
          <Item
            label="100"
            value="100"
          />
        </Styled(PickerNB)>
      </Styled(Item)>
    </Styled(Form)>
  </Styled(Content)>
</Styled(Container)>
`;
